### DISCUSSION TITLE
Project Restructure

### DISCUSSION THREAD

**Discussion by robbrad** [robbrad/UKBinCollectionData#62] [Ideas]
Hi have been thinking we need to restructure the Python project to support testing and namely the python path

At the moment the layout is a bit messy. There is a scripts folder and a tests folder outside of that - this is a bit of a hangover from when this was a single script that just did a one off pull

I was thinking the project could take the form of https://realpython.com/python-application-layouts/#installable-single-package instead

<img width="737" alt="image" src="https://user-images.githubusercontent.com/6122503/180756491-7b1a4247-bb01-437d-b4b7-92be7fd0c505.png">

This would be an opportunity to clean up some of the naming etc

Also this should help with the python paths and the importing of resources for testing that is hard work on #53

**Comment by robbrad** [comment]
@dp247 - what do you think ? I think you might have more experience than me here

**Reply by dp247** [reply]
> @dp247 - what do you think ? I think you might have more experience than me here

I've... actually never built a python package before, but I can see the benefits of having the project formatted like it is in the image.

> This would be an opportunity to clean up some of the naming etc

I do think this is an excellent point though, especially my recent contributions, I think I should have named them fully lol

**Comment by robbrad** [comment]
Getting there https://github.com/robbrad/UKBinCollectionData/tree/feature/%2353_integration_tests 😅

**Comment by robbrad** [comment]
This will answer #60 #53 #56 #64

**Comment by robbrad** [comment]
@dp247  https://github.com/robbrad/UKBinCollectionData/actions/runs/2776647785 SWEEEEET

**Reply by robbrad** [reply]
and https://app.codecov.io/gh/robbrad/UKBinCollectionData !

**Reply by dp247** [reply]
NICEEEEEEEEEEEEE

**Comment by robbrad** [comment]
Just need some help now writing / making the features for each we have already

Then Ill set the GH action to run nightly which should validate if the councils continue to work
